site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md

```diff
@@ -97,7 +97,52 @@ Swagger 2.x compatibility.
 
 ## AQL
 
-
+### Filter conditions in `UPSERT` operations (experimental)
+
+Version 3.12 introduces an alternative syntax for `UPSERT` operations that
+allows you to use dynamic attribute names to look up documents. Previously,
+the expression used to look up a document had to be an object literal.
+
+{{< tabs groupid="UPSERT syntax" >}}
+
+{{< tab name="Exact-value matching" >}}
+```aql
+UPSERT <lookup-document>
+INSERT ...
+UPDATE|REPLACE ...
+IN <collection>
+```
+{{< /tab >}}
+
+{{< tab name="Using FILTER conditions" >}}
+```aql
+UPSERT FILTER <filter-condition>
+INSERT ...
+UPDATE|REPLACE ...
+IN <collection>
+```
+{{< /tab >}}
+
+{{< /tabs >}}
+
+The arbitrary filter condition for the lookup can make use of the pseudo-variable
+`CURRENT`, and it can access the document and apply more filters on it than just
+equality matches. Example:
+
+```aql
+UPSERT FILTER CURRENT.category == 'test && CURRENT.dateTime >= ... && CURRENT.dateTime < ... && CURRENT.status != 'inactive'
```

Comment:
`CURRENT.dateTime >= ... && CURRENT.dateTime < ...` should potentially be substituted by `IN_RANGE(CURRENT.dateTime, ... , ... , true, false)` and concrete values would be nice.
```suggestion
UPSERT FILTER CURRENT.category == 'test' && CURRENT.dateTime >= ... && CURRENT.dateTime < ... && CURRENT.status != 'inactive'
```